packages/dev/core/src/DeviceInput/eventFactory.ts

```diff
@@ -70,7 +72,7 @@ export class DeviceEventFactory {
             evt.pointerType = "mouse";
         } else {
             evt.deviceType = DeviceType.Touch;
-            evt.pointerId = deviceSlot;
+            evt.pointerId = pointerId || deviceSlot;

```

Comment:
pointerId might be 0. Check for the type of pointerId instead of a simple `||`, or use `??`

Reply:
Thanks for catching that!  I tend to forget sometimes that 0 evaluates to false.